app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -104,6 +122,28 @@ export default Component.extend({
     quoteState.selected(postId, _selectedText, opts);
     this.set("visible", quoteState.buffer.length > 0);
 
+    if (this.siteSettings.enable_fast_edit) {
+      const regexp = new RegExp(quoteState.buffer, "gi");
```

Comment:
I think the regex should be escaped first.

Reply:
yes it was on my todo list I noticed issues which might be resolved by this